hollow/src/main/java/com/netflix/hollow/core/read/engine/list/HollowListTypeReadStateShard.java

```diff
@@ -108,9 +108,14 @@ protected void applyToChecksum(HollowChecksum checksum, BitSet populatedOrdinals
                 checksum.applyInt(ordinal);
                 for(int i=0;i<size;i++)
                     checksum.applyInt(getElementOrdinal(shardOrdinal, i));
-            }
 
-            ordinal = populatedOrdinals.nextSetBit(ordinal + 1);
+                ordinal = ordinal + numShards;
+            } else {
+                // Round up ordinal
+                int r = (ordinal & -numShards) + shardNumber;
+                ordinal = (r <= ordinal) ? r + numShards : r;
+            }
+            ordinal = populatedOrdinals.nextSetBit(ordinal);
```

Comment:
Shouldn't this be `ordinal = populatedOrdinals.nextSetBit(ordinal+1);`?

Reply:
Oh, I see that you are incrementing by `numShards` in the matched shard case above.  So most of the time, you're doing the rounding in the next iteration of the loop after this call.  
Might it be clearer to round/align the ordinal with the shard here, rather than in `(104,112-117)`?

Reply:
I am not following you. The calculation at line 112, `ordinal = ordinal + numShards`, is just an optimized version of that on lines 114-116. Both round up to align to the next ordinal to check if set.

A simplification could be:
```
if((ordinal & (numShards - 1)) == shardNumber) {
    ...
}
// Round up ordinal to align with shard number
int alignedOrdinal = (ordinal & -numShards) + shardNumber;
alignedOrdinal = (r <= ordinal) ? r + numShards : r;
ordinal = populatedOrdinals.nextSetBit(alignedOrdinal)
```
I doubt that would impact the performance.